text/0000-more-api-documentation-conventions.md

```diff
@@ -0,0 +1,409 @@
+- Feature Name: More API Documentation Conventions
+- Start Date: 2016-03-31
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+[RFC 505] introduced certain conventions around documenting Rust projects. This RFC supersedes
+that one, though it has the same aims: to describe how the Rust project should be documented,
+and provide guidance for other Rust projects as well.
+
+This RFC will contain some similar text as RFC 505, so that we can have one RFC with the full
+conventions.
+
+[RFC 505]: https://github.com/rust-lang/rfcs/blob/master/text/0505-api-comment-conventions.md
+
+# Motivation
+[motivation]: #motivation
+
+Documentation is an extremely important part of any project. It’s important
+that we have consistency in our documentation.
+
+For the most part, the RFC proposes guidelines that are already followed today,
+but it tries to motivate and clarify them.
+
+# Detailed design
+[design]: #detailed-design
+
+This RFC is large. Here’s a table of contents:
+
+* [Content](#content)
+  * [Summary sentence](#summary-sentence)
+  * [English](#english)
+* [Form](#form)
+  * [Use line comments](#use-line-comments)
+  * [Using Markdown](#using-markdown)
+* [Example](#example)
+
+## Content
+[content]: #content
+
+These conventions relate to the contents of the documentation, the words themselves.
+
+### Summary sentence
+[summary-sentence]: #summary-sentence
+
+In API documentation, the first line should be a single-line short sentence
+providing a summary of the code. This line is used as a summary description
+throughout Rustdoc’s output, so it’s a good idea to keep it short.
+
+The summary line should be written in third person singular present indicative
+form. Basically, this means write “Returns” instead of “Return.”
```

Comment:
This might be a little off-topic—and maybe nit-picky—but here goes.

One thing that bugs me about Rust documentation—including the documentation I write for my crate—is that the summary line looks ugly. In a module's documentation page, where the types, functions, methods, etc. are neatly aligned in tables, the summary line looks best as a fragment. E.g.,

<table>
<tr><td><code>Foo</code></td><td>Manages a foobar thingamabob.</td></tr>
<tr><td><code>Qux</code></td><td>Wraps a blah blah blah.</td></tr>
</table>

But when I navigate to the documentation page for `Foo` , the same fragment looks _abrupt_ because it's in prose. Instead, I would like the summary to be a complete sentence starting with a noun. E.g., <q>`Foo` manages a foobar thingamabob.</q> 

I wish Rust automatically formatted this for me. That is, I wish I could write a complete sentence for the summary line and have `rustdoc` leave out the noun when generating the one-liner for the module-level table. I suspect `rustdoc` wouldn't need to be very smart to do this. It could recognize that the sentence starts with <q>Foo</q> and, because `Foo` is the thing being documented, elide it from the output and capitalize the following word. E.g., <q>`Foo` manages</q> becomes <q>Manages.</q>

In addition to looking prettier, I think this would also encourage documentation authors to write better summary lines.